### ISSUE TITLE
[scoped-custom-element-registry] No support for append

### ISSUE THREAD

**Issue by jonathandewitt-dev** [webcomponents/polyfills#600]
### Description

Currently, the only way I can get scoped elements to upgrade is by using `innerHTML`, but if I attempt with `append`, nothing upgrades.

### Example

This works:
```ts
shadowRoot.innerHTML = '<scoped-element></scoped-element>';
```
This does not:
```ts
shadowRoot.append(fragment);
```

#### Expected behavior

Element should upgrade when appended.

#### Actual behavior

Element is not upgraded.

### Version

0.0.9

### Browsers affected

- [x] Chrome
- [x] Firefox
- [x] Edge
- [x] Safari
- [x] IE 11

**Comment by sorvell** [comment]
It matters how the `fragment` in question was created. Per the current proposal, the fragment must be created via a shadowRoot in order for the elements to customize. Here is a [working example](https://lit.dev/playground/#gist=236cc0e3f588496ac66f75bdbf08b86d). 

If you are doing this and it's still not working, we'll need more info and preferably a reproduction to help.

**Comment by sorvell** [comment]
Just wanted to add that there can be considerable nuance here of which to be aware. Elements in fragments created not using `shadowRoot.importNode` can sometimes upgrade as expected when appended, but this is unreliable. 

This works when:

1. the fragment is not owned by the document, e.g. was created via `template.content.cloneNode(true)`
2. or it was created with elements before they were defined and never connected to the document.

**Comment by jonathandewitt-dev** [comment]
You know what, I feel kind of silly after looking at your working example.

My experience with fragments is backwards compared to most devs, because I found out about them before templates. So the way I was creating a fragment was based on knowledge that is probably borderline irrelevant today. This was my `parseFragment()` function:

```ts
const parseFragment = (htmlStr: string): DocumentFragment => {
	const range = document.createRange();
	range.selectNode(document.body); // required in Safari
	return range.createContextualFragment(htmlStr);
};
```

Simply changing that to this solved this issue:
```ts
const parseFragment = (htmlStr: string): DocumentFragment => {
	const template = document.createElement('template');
	template.innerHTML = htmlStr;
	return template.content;
};
```

And I did know this was essentially the same result but simpler, yet I didn't even think about it until I looked at your code. One of those things, I guess.

That being said, that little `importNode` nugget will undoubtedly be useful going forward, so thanks for the heads up. I'll close this issue.
